### PR TITLE
Issue/#5 initialize backend springboot

### PR DESCRIPTION
**Pull request by razine-bensari**
# Related Issue
- https://github.com/DPigeon/Money-Tree/issues/5

# Proposed changes
- Switched to Java backend with Spring
- For testing we will be using Spock with Groovy. Groovy is a superset of java (aka java code can be written inside groovy files) it is much easier to write Groovy than java for test, and all documentation in Spock is in groovy. I wrote a sample test file to demonstrate the feature of Spock with Groovy/Java

# Reviewer(s)
 - @Alessandro100 
- @abdulmansour 
- @EspressoCode

### CODE REVIEW

**Review by EspressoCode — APPROVED** [review]
Looks good to me. I would suggest adding more to the readme file to elaborate on our backend technology and to help all members to get familiar with it in future commits. Very good work and research 👍🏻

File: backend/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# How to run locally
+
+## JAVA
+
+You need java 11 to run the application.
```
Comment (Alessandro100) [inline backend/README.md:5]: you can mention that you can check your java version with `java -version`

File: backend/README.md [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+You need to install neo4j in your OS of choice. For macOS users, you can use brew by running `brew install neo4j` in your terminal.
+
+- You can use a GUI to look at your local neo4j instance using neo4j desktop (https://neo4j.com/download-neo4j-now/).
+
+- When you first start neo4j, the default username and password is neo4j (for both). neo4j will not allow any application
+to connect if the password has not been changed.
+
+- To start neo4j locally, open your terminal and run `neo4j start`
+
+- If neo4j is up and running locally, you can run `curl -v -u neo4j:neo4j -X POST localhost:7474/user/neo4j/password -H "Content-type:application/json" -d "{\"password\":\"secret\"}"` and replace the secret string with stonecap. stonecap is our db password for dev when running the application locally.
```
Comment (Alessandro100) [inline backend/README.md:28]: I would add the password directly into the code like 
`curl -v -u neo4j:neo4j -X POST localhost:7474/user/neo4j/password -H "Content-type:application/json" -d "{\"password\":\"stonecap\"}"`

so people can easily copy paste it. You can also mention that if this doesn't work, to go through the browser at 'localhost:7474'

**Review by Alessandro100 — APPROVED** [review]
Great stuff! It's appreciated that you went above and beyond by integrating Neo4j. I left a few comments on improving the README, but its minor stuff. Hyped to have this integrated

**Comment by DPigeon** [comment]
Looks good! Thanks for updating the README file.

**Pull request merged**